mslib/mswms/mss_plot_driver.py

```diff
@@ -676,3 +677,241 @@ def plot(self):
+        self.lsec_path = lsec_path
+        self.lsec_numpoints = lsec_numpoints
+        self.lsec_path_connection = lsec_path_connection
+
+    def _load_interpolate_timestep(self):
+        """
+        Load and interpolate the data fields as required by the linear
+        section style instance. Only data of time <fc_time> is processed.
+
+        Shifts the data fields such that the longitudes are in the range
+        left_longitude .. left_longitude+360, where left_longitude is the
+        westmost longitude appearing in the list of waypoints minus one
+        gridpoint (to include all waypoint longitudes).
+
+        Necessary to prevent data cut-offs in situations where the requested
+        cross section crosses the data longitude boundaries (e.g. data is
+        stored on a 0..360 grid, but the path is in the range -10..+20).
+        """
+        if self.dataset is None:
+            return {}
+        data = {}
+
+        timestep = self.times.searchsorted(self.fc_time)
+        logging.debug("loading data for time step %s (%s)", timestep, self.fc_time)
+
+        # Determine the westmost longitude in the cross-section path. Subtract
+        # one gridbox size to obtain "left_longitude".
+        dlon = self.lon_data[1] - self.lon_data[0]
+        left_longitude = self.lons.min() - dlon
+        logging.debug("shifting data grid to gridpoint west of westmost "
+                      "longitude in path: %.2f (path %.2f).",
+                      left_longitude, self.lons.min())
+
+        # Shift the longitude field such that the data is in the range
+        # left_longitude .. left_longitude+360.
+        # NOTE: This does not overwrite self.lon_data (which is required
+        # in its original form in case other data is loaded while this
+        # file is open).
+        lon_data = ((self.lon_data - left_longitude) % 360) + left_longitude
+        lon_indices = lon_data.argsort()
+        lon_data = lon_data[lon_indices]
+        factors = []
+
+        for name, var in self.data_vars.items():
+            data[name] = []
+            if len(var.shape) == 4:
+                var_data = var[timestep, ::-self.vert_order, ::self.lat_order, :]
+            else:
+                var_data = var[:][timestep, np.newaxis, ::self.lat_order, :]
+            logging.debug("\tLoaded %.2f Mbytes from data field <%s> at timestep %s.",
+                          var_data.nbytes / 1048576., name, timestep)
+            logging.debug("\tVertical dimension direction is %s.",
+                          "up" if self.vert_order == 1 else "down")
+            logging.debug("\tInterpolating to cross-section path.")
+            # Re-arange longitude dimension in the data field.
+            var_data = var_data[:, :, lon_indices]
+
+            points = (self.vert_data, self.lat_data, lon_data)
+            inter_points = np.array([[self.alts[i], self.lats[i], self.lons[i]] for i in range(len(self.lats))])
```

Comment:
I would leverage the vertical section interpolation first and then interpolate (np.interp) vertically to a 1-D line. 
One might think about interpolating in log-pressure instead of pressure.
Should be fine performance-wise as well and we have less different interpolations.